tests/utils.py

```diff
@@ -87,3 +87,21 @@ def check_server_ready(protocol="grpc"):
         output = buf.getvalue()
     output = json.loads(output)
     return output["ready"]
+
+
+# Context Manager to start and kill a mock server running in background and used by testing functions
```

Comment:
Please also update title and description to remove mentions of "mock" - these aren't mocking
```suggestion
# Context Manager to start and kill a server running in background and used by testing functions
```